scheduler/src/cook/mesos/share.clj

```diff
@@ -146,13 +175,11 @@
                                  :where
                                  [?q :share/user ?user]]
                                db)
-          user->share-cache (->> all-share-users
-                                 (map (fn [user]
-                                        [user (get-share db user)]))
-                                 ;; In case default-user doesn't have an explicit share
-                                 (cons [default-user (get-share db default-user)])
-                                 (into {}))]
+          default-user-share (get-share db default-user)
+          user->share-cache (-> (get-shares db all-share-users)
+                                ;; In case default-user doesn't have an explicit share
+                                (assoc default-user default-user-share))]
       (fn user->share
```

Comment:
This is indirectly called from `rank-jobs`. So optimizing this will also help speed up the ranking.